common/benchmark_main.cpp

```diff
@@ -0,0 +1,33 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <benchmark/benchmark.h>
+
+#include "absl/flags/parse.h"
+#include "common/init_llvm.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/StringRef.h"
+
+auto main(int orig_argc, char** orig_argv) -> int {
+  // Inject a flag to override the defaults for benchmarks. This can still be
+  // disabled by user arguments.
+  llvm::MutableArrayRef<char*> orig_argv_ref(orig_argv, orig_argc);
+  int argc = orig_argc + 1;
+  llvm::OwningArrayRef<char*> injected_argv_storage(argc + 2);
+  injected_argv_storage[0] = orig_argv[0];
+  char injected_flag[] = "--benchmark_counters_tabular";
+  injected_argv_storage[1] = injected_flag;
```

Comment:
The pedant in me wants to point out that this doesn't do the right thing if `orig_argc` is 0. But I don't think we need to care about adversarial calls to `exec` :-)